src/AzureReadStream.cs

```diff
@@ -198,9 +234,26 @@ private void LoadSyncBuffer()
 
             _bufferOffset = 0;
             _bufferEnd = (int)Math.Min(_buffer.Length, Length - _position);
-
             AzureRetry.Do(
-                c => _blob.DownloadRangeToByteArrayAsync(_buffer, 0, _position, _bufferEnd, c),
+                async c =>
+                {
+                    var downloadUrl = _blob.GenerateSasUri(new BlobSasBuilder(BlobContainerSasPermissions.Read,
+                                                           new DateTimeOffset(DateTime.UtcNow.AddDays(1))));
+                    using (var httpClient = new HttpClient())
+                    {
+                        var bytesEnd = _position < _bufferEnd ? $"{_bufferEnd - 1}" : "";
+                        var range = $"bytes={_position}-{bytesEnd}";
```

Comment:
`_bufferEnd` is relative to the `_buffer`, so it cannot be used for the range. I think this would be `{position}-{position + _bufferEnd - 1}` which, based on the above, should always be a valid range.